### PR TITLE
host: Allow comments in JSON files

### PR DESCRIPTION
**Pull request by lpereira**
Although JSON doesn't support comments, a common extension is to enable
JavaScript-style comments.  By default, RapidJSON will error out if
comments are found in the input data, but it can be configured to
ignore them.

This is useful especially when testing out changes in
runtimeconfig.json files.

Fixes dotnet/core-setup#5661

### CODE REVIEW

**Review by vitek-karas — COMMENTED** [review]
Please add a test to verify it works as expected. Should be trivial as we have lot of runtimeconfig parsing tests already present.

**Comment by am11** [comment]
Is it possible for host to call managed API using `ICLRRuntimeHost::ExecuteInDefaultAppDomain()` etc.?
If so, RapidJSON clone can be removed in favor of `System.Text.JSON`.

**Comment by jkoritzinsky** [comment]
At the time we’re parsing the runtimeconfig.json we still haven’t discovered which coreclr.dll we’re going to use, so we can’t call into any managed code.

**Comment by lpereira** [comment]
@vitek-karas wrote:

 > Please add a test to verify it works as expected. Should be trivial as we have lot of runtimeconfig parsing tests already present.

Tests have been added.  Could you please take a look again?

**Review by vitek-karas — APPROVED** [review]
:shipit:

**Review by elinor-fung — APPROVED** [review]

File: src/installer/test/Assets/TestUtils/SDKLookup/dotnet.runtimeconfig.json
Comment (elinor-fung) [inline src/installer/test/Assets/TestUtils/SDKLookup/dotnet.runtimeconfig.json]: As a side note, I think the tests that use this right now don't strictly need to do the config parsing, since they are just for validating SDK lookup - I would argue they shouldn't once we have a mockhostfxr (and so the comment validation would need to be moved).

**Comment by lpereira** [comment]
Rebased the PR on top of current master.  Will wait until CI is happy before merging.

**Pull request merged**
